drivers/i3c/i3c_cdns.c

```diff
@@ -1756,23 +1782,36 @@ static int cdns_i3c_i2c_api_configure(const struct device *dev, uint32_t config)
  */
 static int cdns_i3c_configure(const struct device *dev, enum i3c_config_type type, void *config)
 {
-	struct cdns_i3c_data *data = dev->data;
-	struct i3c_config_controller *ctrl_cfg = config;
+	if (type == I3C_CONFIG_CONTROLLER) {
+#ifdef CONFIG_I3C_CONTROLLER
+		struct cdns_i3c_data *data = dev->data;
+		struct i3c_config_controller *ctrl_cfg = config;
 
-	if ((ctrl_cfg->scl.i2c == 0U) || (ctrl_cfg->scl.i3c == 0U)) {
-		return -EINVAL;
-	}
+		if ((ctrl_cfg->scl.i2c == 0U) || (ctrl_cfg->scl.i3c == 0U)) {
+			return -EINVAL;
+		}
 
-	data->common.ctrl_config.scl.i3c = ctrl_cfg->scl.i3c;
-	data->common.ctrl_config.scl.i2c = ctrl_cfg->scl.i2c;
+		data->common.ctrl_config.scl.i3c = ctrl_cfg->scl.i3c;
+		data->common.ctrl_config.scl.i2c = ctrl_cfg->scl.i2c;
 
-	k_mutex_lock(&data->bus_lock, K_FOREVER);
-	cdns_i3c_set_prescalers(dev);
-	k_mutex_unlock(&data->bus_lock);
+		k_mutex_lock(&data->bus_lock, K_FOREVER);
+		cdns_i3c_set_prescalers(dev);
+		k_mutex_unlock(&data->bus_lock);
+#else
+		return -EINVAL;
+#endif
+	} else if (type == I3C_CONFIG_TARGET) {
+#ifdef CONFIG_I3C_TARGET
+		return 0;
+#else
+		return -EINVAL;
+#endif
```

Comment:
more a question here.  Is it possible to have a type of `I3C_CONFIG_TARGET` without having that enabled?  Asking, as this small section of code to me suggests it's possible get yourself into a confused state between run-time and compile-time functionality.  I see a few other similar mitigation on this behavior as well.

Reply:
it could still be possible to pass it in as its just an enum, but I probably should be returning -ENOTSUP instead here if that compile option is off